sql/src/main/java/io/crate/planner/operators/Filter.java

```diff
@@ -65,7 +66,17 @@
         return sourceBuilder;
     }
 
-    Filter(LogicalPlan source, QueryClause queryClause) {
+    public static LogicalPlan create(LogicalPlan source, Symbol query) {
+        if (query instanceof Literal) {
+            Boolean value = (Boolean) ((Literal) query).value();
```

Comment:
This is not safe, is it?

Inside Literal we are checking to see if the value is an array at some point value.getClass().isArray()

Reply:
query symbols must always have a boolean type - but I can add an assertion